pkg/controller/podautoscaler/horizontal.go

```diff
@@ -52,15 +52,56 @@ import (
 )
 
 var (
-	scaleUpLimitFactor  = 2.0
-	scaleUpLimitMinimum = 4.0
+	scaleUpLimitFactor              = 2.0
+	scaleUpLimitMinimum             = 4.0
+	scaleUpLimitPercent       int32 = 100
+	scaleUpLimitMinimumPods   int32 = 4
+	scaleUpPeriod             int32 = 60
+	scaleUpDelaySeconds       int32
+	maxPolicy                 = autoscalingv2.MaxPolicySelect
+	defaultHPAScaleUpBehavior = autoscalingv2.HPAScalingRules{
+		StabilizationWindowSeconds: &scaleUpDelaySeconds,
+		SelectPolicy:               &maxPolicy,
+		Policies: []autoscalingv2.HPAScalingPolicy{
+			{
+				Type:          autoscalingv2.PodsScalingPolicy,
+				Value:         scaleUpLimitMinimumPods,
+				PeriodSeconds: scaleUpPeriod,
+			},
+			{
+				Type:          autoscalingv2.PercentScalingPolicy,
+				Value:         scaleUpLimitPercent,
+				PeriodSeconds: scaleUpPeriod,
+			},
+		},
+	}
+	scaleDownPeriod             int32 = 60
+	scaleDownDelaySeconds       int32 = 300
+	scaleDownLimitPercent       int32 = 100
+	defaultHPAScaleDownBehavior       = autoscalingv2.HPAScalingRules{
+		StabilizationWindowSeconds: &scaleDownDelaySeconds,
+		SelectPolicy:               &maxPolicy,
+		Policies: []autoscalingv2.HPAScalingPolicy{
+			{
+				Type:          autoscalingv2.PercentScalingPolicy,
+				Value:         scaleDownLimitPercent,
+				PeriodSeconds: scaleDownPeriod,
+			},
+		},
```

Comment:
What existing limit is this policy replacing?  What effect will this have?

I know that the two scale up policies are replacing existing constants (limit 2x or 4 pods, whichever is larger) and the scale down stabilization window is 300 seconds.  But I'm not aware of an existing scale down limit of 100% within 60 seconds which requires this default scale down policy.

Reply:
This is how it works:
1. If the user doesn't specify any policy, we use the default one (that is listed here).
2. It is needed because if no policy is specified, the HPA will not scale down as it doesn't have any policy to follow:
https://github.com/kubernetes/kubernetes/pull/74525/files/0328cec975e6438009d2b9a069a034a63f9df1b7#diff-ff4e8f1a0c24ee394a0ddf37d64b2960R1098

So, the default policy allows scaling down to 0%.